cmd/influx/main.go

```diff
@@ -176,12 +176,12 @@ func (b *cmdInfluxBuilder) cmd(childCmdFns ...func(f *globalFlags, opt genericCL
 
 	if flags.Token == "" {
 		// migration credential token
-		migrateOldCredential()
-
-		// this is after the flagOpts register b/c we don't want to show the default value
-		// in the usage display. This will add it as the config, then if a token flag
-		// is provided too, the flag will take precedence.
-		flags.Config = getConfigFromDefaultPath()
+		if migrateOldCredential() {
+			// this is after the flagOpts register b/c we don't want to show the default value
+			// in the usage display. This will add it as the config, then if a token flag
+			// is provided too, the flag will take precedence.
+			flags.Config = getConfigFromDefaultPath()
```

Comment:
This change will break the multiple configs feature. Usage display override is related to the cobra persistent flags, has nothing to do with migration. https://github.com/influxdata/influxdb/issues/17259

Reply:
Understood, I am going to revert this part to a state "before this PR".